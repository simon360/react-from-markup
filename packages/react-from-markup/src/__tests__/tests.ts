/* eslint-env jest */
import * as React from "react";
import reactFromMarkupContainer from "..";

describe("reactFromMarkupContainer E2E tests", async () => {
  it("Should rehydrate a basic component", async () => {
    const COMPONENT_NAME: string = "myComponent";
    let myComponentRef: HTMLElement | null = null;

    const rehydrator = async () => {
      return React.createElement(
        "span",
        {
          ref: ref => {
            myComponentRef = ref;
          },
          testid: COMPONENT_NAME
        },
        "rehydrated component"
      );
    };

    const rehydrators = { [COMPONENT_NAME]: rehydrator };
    const documentElement = document.createElement("div");

    documentElement.innerHTML = `
      <div data-react-from-markup-container>
        <div data-rehydratable="${COMPONENT_NAME}"></div>
      </div>`;

    await reactFromMarkupContainer(documentElement, rehydrators, {
      extra: {}
    });

    expect(documentElement.querySelector(`[testid=${COMPONENT_NAME}]`)).toBe(
      myComponentRef
    );
    expect(documentElement.innerHTML).toMatchSnapshot();
  });

  it("Should rehydrate valid HTML markup", async () => {
    const documentElement = document.createElement("div");

    documentElement.innerHTML = `
    <div data-react-from-markup-container>
      <p style="">paragraph</p>
      <p ref="">paragraph</p>
      <p key="">paragraph</p>
      <p children="">paragraph</p>
      <p dangerouslySetInnerHTML="{__html:'should not be children'}"></p>
    </div>`;

    await reactFromMarkupContainer(documentElement, {}, { extra: {} });

    expect(documentElement.innerHTML).toMatchSnapshot();
  });
});
