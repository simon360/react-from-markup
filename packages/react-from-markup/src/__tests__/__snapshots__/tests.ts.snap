// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reactFromMarkupContainer E2E tests Should rehydrate a basic component 1`] = `
"
      <div data-react-from-markup-container=\\"\\">
        <span testid=\\"myComponent\\">rehydrated component</span>
      </div>"
`;
